packages/context/src/inject.ts

```diff
@@ -201,9 +203,31 @@ export namespace Getter {
 }
 
 /**
- * The function injected by `@inject.setter(key)`.
+ * The function injected by `@inject.setter(bindingKey)`. It sets the underlying
+ * binding to a const value. Returns the `Binding` object so that the binding
+ * can be further configured, such as setting it to a class with `toClass()`.
```

Comment:
I am not convinced it's a good idea to return the binding from the setter. I don't want us to leak details about our IoC container (Context/Binding classes) to code receiving dependencies. As I wrote above, I see `Setter` as a concept that's agnostic of DI/IoC and I'd like to preserve that.

For use cases where the developer needs to set more than just the bound value, there is the new `@inject.binding` decorator to give full control over the binding.

I am also concerned about robustness of the proposed solution. It allows developers to invoke the injected setter function without any value provided:

```ts
setter(); 
```

What's the expected outcome? For example, if `T` is `string | undefined`, the call above should be valid and should set the target binding value to `undefined`.

For the reasons outline above, I am against changing the signature of `Setter` function and propose to keep it in the current form.

Reply:
I can revert the return type back to `void`. What about the `bindingCreation` option?

Reply:
reverted.